src/backend/drm/Renderer.hpp

```diff
@@ -24,18 +24,22 @@ namespace Aquamarine {
     class CDRMRendererBufferAttachment : public IAttachment {
       public:
         CDRMRendererBufferAttachment(Hyprutils::Memory::CWeakPointer<CDRMRenderer> renderer_, Hyprutils::Memory::CSharedPointer<IBuffer> buffer, EGLImageKHR image, GLuint fbo_,
-                                     GLuint rbo_, SGLTex tex);
+                                     GLuint rbo_, SGLTex tex, uint8_t* intermediateBuf, size_t intermediateBufLen);
         virtual ~CDRMRendererBufferAttachment() {
             ;
         }
         virtual eAttachmentType type() {
             return AQ_ATTACHMENT_DRM_RENDERER_DATA;
         }
 
-        EGLImageKHR                                   eglImage = nullptr;
-        GLuint                                        fbo = 0, rbo = 0;
-        SGLTex                                        tex;
-        Hyprutils::Signal::CHyprSignalListener        bufferDestroy;
+        EGLImageKHR                            eglImage = nullptr;
+        GLuint                                 fbo = 0, rbo = 0;
+        SGLTex                                 tex;
+        Hyprutils::Signal::CHyprSignalListener bufferDestroy;
+        // This is malloc'd manually instead of using a std::vector to keep lifetime management in line with the rest of the class,
+        // which e.g. doesn't immediately free the eglImage on drop but instead waits until onBufferAttachmentDrop.
+        uint8_t*                                      intermediateBuf    = nullptr;
```

Comment:
that's completely unnecessary. You can use a vector and reset it in the drop() handler.

Reply:
I was worried about it getting freed early, not freed late. But yeah I can just replace this with a vector. I don't think I need to reset it in the drop handler because that's only called from the destructor of the buffer which should call the destructor of the attachments immediately afterwards.

Reply:
I've gone ahead and refactored this to use std::vector and std::span for the intermediateBuf. I'm coming from rust so let me know if any of this isn't idiomatic and I'd be happy to fix it (e.g. I debated using a std::optional for intermediateBuf but I figured it'd be better to just check if it's empty in C++).